### PR TITLE
[ENH][RFC] Add deprecation helper

### PR DESCRIPTION
**Pull request by massich**
Changing the render class would lead to deprecate lots of parameters (#5822).
Therefore, with @GuillaumeFavelier we came up with this decorator.

I guess it would be good to integrate all the deprecation objects etc..
but this is a first step (and we think it would do the job).

WDYT??

cc: @mne-tools/contributors

### CODE REVIEW

File: mne/utils.py [hunk 1]
```diff
@@ -47,6 +47,8 @@
 
 from .fixes import _get_args
 
+from mne import __version__
```
Comment (larsoner) [inline mne/utils.py]: avoid absolute imports, instead in the function that needs it nest a `from . import __version__`

**Comment by larsoner** [comment]
Seems reasonable to me. But then again I think #5753 is reasonable, too :)

**Comment by massich** [comment]
What is not covered in this PR is to introspect the docstring to add it in the documentation.

File: mne/utils.py [hunk 2]
```diff
@@ -899,6 +901,63 @@ def _update_doc(self, olddoc):
...
+                               details=None,
+                               transform=None):
+    """Mark a function parameter as deprecated.
+
+    Issue a warning when the function is called with the `old_parameter`.
+
+
+    Parameters
+    ----------
+    extra: string
```
Comment (jasmainak) [inline mne/utils.py]: corresponding argument is missing

**Comment by kchawla-pi** [comment]
Hey @massich @GuillaumeFavelier this is pretty cool, not sure how you kept the code compact, will study and learn!

I think instead of using keyword args `old_parameter`, `new_parameter`, the function should accept a dict of them, such as
`{old_param0: new_param0, old_param1: new_param1, old_param2: new_param2}'.
This way one can deprecate multiple parameters in a function if needed and seems cleaner to me.

**Comment by kchawla-pi** [comment]
I mean I wrote a decorator for the same purpose but it is not so compact as this one. And I haven't adapted it as a utility function yet, I will in time.

**Comment by massich** [comment]
What do we do here? @GuillaumeFavelier are you still interested in this one? If so I can finish it quickly.

@larsoner 	thoughts?

**Comment by kchawla-pi** [comment]
We have also implemented this functionality in Nilearn & Nistats and it is recurringly useful, currently being used in 5 places..

**Comment by GuillaumeFavelier** [comment]
In the end, we came up with a different strategy and the deprecation was not needed anymore in #5822 

I still think it could be useful in some cases or even just for maintenance reasons.

**Comment by kchawla-pi** [comment]
@GuillaumeFavelier So Sicch tried so hard and got so far, but in the end it doesn't even matter??

**Comment by kchawla-pi** [comment]
Which PR is that? I would like to see the approach you used to solve the root thing.

**Comment by GuillaumeFavelier** [comment]
I wouldn't say it doesn't matter because as I told you I think the value of our efforts can be leveraged somewhere else hopefully.

The deprecation matter arose as I tried to replace the `figure` parameter but during the discussion, we decided not to deprecate at all.

File: mne/utils.py [hunk 3]
```diff
@@ -899,6 +901,63 @@ def _update_doc(self, olddoc):
...
+                 'new_param': new_param,
+                 'details': details}
+    _MSG = ('`{old_param}` is deprecated and will be replaced by `{new_param}`'
+            ' in {deprecated_in}. `{old_param}` will be no longer present'
+            ' starting in {removed_in}. {details}').format(**_MSG_KEYS)
+
+    # XXX: __version__ should be tested
+
+    def true_decorator(f):
+        @wraps(f)
```
Comment (larsoner) [inline mne/utils.py]: `wraps` is not be enough to get the docstring correct here. You need to use the `externals/decorator` functionality.

**Comment by kchawla-pi** [comment]
decorator is necessary for Python 2. Is that still supported?

**Comment by larsoner** [comment]
IIRC even in Python 3 `wraps` does not fix all problems with wrapping and docstring parsing

**Comment by larsoner** [comment]
@massich think this is worth coming back to, or should we close?

**Comment by massich** [comment]
I took part of it for my PR depreciating montage, but it was hard to review
and we took a diferent approach.

If guillaume still needs it I'll finish it. Otherwise I think I'll close
it.

**Comment by GuillaumeFavelier** [comment]
I don't need it for now @massich , feel free to close it. we can still reopen later if the situation changes

**Pull request closed without merging**